### PR TITLE
INTMDB-562: Fixing some issues with autogen

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR fixes the following issues that I encountered while using schema-gen:

- the generated JSON file contains empty definitions and properties
- the logic was considering only response 200 and skipping response 201
- the logic was adding programmatic API keys to all the views 

Issue: #205

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change works in Atlas

## Further comments

I tested my changes by creating the JSON schema for the cluster and project resource.

### CODE REVIEW

File: cfn-resources/autogen/schema-gen/main.go [hunk 1]
```diff
@@ -282,28 +277,15 @@ func readArgs() (compare bool, err error) {
...
-		props[key] = properties[key]
-	}
-	return props
-}
-
-func sortDefinitions(properties map[string]Definitions) (props map[string]Definitions) {
-	keys := make([]string, len(properties))
-	for key := range properties {
-		keys = append(keys, key)
+func sortProperties[V any](properties map[string]V) (props map[string]interface{}) {
```
Comment (andreaangiolillo) [inline cfn-resources/autogen/schema-gen/main.go:280]: using generics instead

File: cfn-resources/autogen/schema-gen/main.go [hunk 2]
```diff
@@ -412,21 +394,34 @@ func readRequestBody(method *openapi3.Operation, doc *openapi3.T) (schemaKeys []
...
-		method.Responses["200"].Value.Content["application/json"].Schema != nil {
-		resSchemaKey = filepath.Base(method.Responses["200"].Value.Content["application/json"].Schema.Ref)
-		// Read from Request body
-		if doc.Components.Schemas[filepath.Base(resSchemaKey)] != nil {
-			value := *doc.Components.Schemas[filepath.Base(resSchemaKey)]
-			resSchema, definitions = processSchema(resSchemaKey, &value, doc.Components.Schemas)
-		}
+	if methodResponseHasSchema(method, "201") {
+		return readResponseBodyWithResponseCode(openAPIDoc, method, "201")
+	}
```
Comment (andreaangiolillo) [inline cfn-resources/autogen/schema-gen/main.go:404]: here if the **responseBody** does not have response 200, we check for response 201 (resource created)

**Review by Zuhairahmed — APPROVED** [review]
LGTM

**Review by maastha — APPROVED** [review]
LGTM

File: cfn-resources/autogen/schema-gen/main.go (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/autogen/schema-gen/main.go:279]: [V any] allows me to use this function for both Definitions and Properties. Getting this error when trying to use `map[string]any`. Any suggestions on how to fix it? 
Screenshot 2023-03-20 at 09 01 48

Comment (andreaangiolillo) [reply cfn-resources/autogen/schema-gen/main.go:279]: I will open a follow-up PR to address this comment if needed. Thanks!

**Pull request merged**